saleor/checkout/complete_checkout.py

```diff
@@ -354,7 +354,7 @@ def _create_order_line_discounts(
 
 def _get_sale_id(line_discounts: list[OrderLineDiscount]):
     for discount in line_discounts:
-        if discount.type == DiscountType.PROMOTION:
+        if discount.type in [DiscountType.CATALOGUE_PROMOTION, DiscountType.PROMOTION]:
```

Comment:
I would suggest to add temporary tests to make sure that backward compatibility is ensured correctly in current version (that means you can handle both discount types)